task-sdk/src/airflow/sdk/bases/operator.py

```diff
@@ -1022,7 +1022,9 @@ def __init__(
     ):
         # Note: Metaclass handles passing in the DAG/TaskGroup from active context manager, if any
 
-        self.task_id = task_group.child_id(task_id) if task_group else task_id
+        # Only apply task_group prefix if this operator was not created from a mapped operator
+        # Mapped operators already have the prefix applied during their creation
+        self.task_id = task_group.child_id(task_id) if task_group and not self.__from_mapped else task_id
         if not self.__from_mapped and task_group:
             task_group.add(self)
```

Comment:
```suggestion
        if task_group and not self.__from_mapped:
            self.task_id = task_group.child_id(task_id)
            task_group.add(self)
        else:
            self.task_id = task_id
```

Let's make it multiple lines

Reply:
@evgenii-prusov I think we're good to merge this one after this 🙂